tests/test_examples.py

```diff
@@ -34,6 +34,7 @@
 EXCLUDE_EXAMPLES = [
     "cross_validation.py",
     "gradient_accumulation.py",
+    "local_sgd.py",
```

Comment:
While this is accurate, to test your script we should also include a test similar to https://github.com/huggingface/accelerate/blob/daf76a7d385b4c56e82acd653bd9e720084b6dff/tests/test_examples.py#L215-L217, to ensure your script runs fine. That would be all the tests that it would need I believe :)

Reply:
Regarding this one, I am a bit unsure how this is invoked. I add a file `examples/by_feature/local_sgd.py`, I can add a similar function test_local_sgd(self). A couple of Qs though:
1. will it be called automatically?
2. which testargs will be used?

Reply:
It gets called when the example tests are through our CI. `testargs` are whatever need to be used to make your script run with the feature enabled. For instance this reads as:

```bash
accelerate launch examples/by_feature/gradient_accumulation.py
```
(`accelerate launch comes from `self._launch_args`)

Yes, add a similar `test_local_sgd()` file which does this, as we make sure all of our example tests get run. to see when they are fully evoked, see the `Makefile` under `test_examples`

Reply:
You need `test_examples`, which was further down from where you looked at... https://github.com/huggingface/accelerate/blob/main/Makefile#L45-L46

Yes, the core tests ignore the examples as they often require GPUs or take a long time..